ui/src/app/tasks/task-create/task-create.component.spec.ts

```diff
@@ -0,0 +1,162 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { DebugElement } from '@angular/core';
+import { By } from '@angular/platform-browser';
+import { BusyModule } from 'tixif-ngx-busy';
+import { ToastyService } from 'ng2-toasty';
+import { ActivatedRoute } from '@angular/router';
+import { FormsModule, ReactiveFormsModule } from '@angular/forms';
+import { RouterTestingModule } from '@angular/router/testing';
+import { TasksService } from '../tasks.service';
+import { TaskCreateComponent } from './task-create.component';
+import { MockActivatedRoute } from '../../tests/mocks/activated-route';
+import { MockToastyService } from '../../tests/mocks/toasty';
+import { MockTasksService } from '../../tests/mocks/tasks';
+
+describe('TaskCreateComponent', () => {
+  let component: TaskCreateComponent;
+  let fixture: ComponentFixture<TaskCreateComponent>;
+  let de: DebugElement;
+  let el: HTMLElement;
+  let activeRoute: MockActivatedRoute;
+  const toastyService = new MockToastyService();
+  const tasksService = new MockTasksService();
+  const commonTestParams = { id: 'faketask' };
+  const commonTestAppInfos = { faketask: {
+    name: 'fakename',
+    options: [
+      {id: 'fakename.opt1', name: 'opt1', defaultValue: 'val1'}
+    ]}};
+
+
+  beforeEach(async(() => {
+    activeRoute = new MockActivatedRoute();
+    TestBed.configureTestingModule({
+      declarations: [
+        TaskCreateComponent
+      ],
+      imports: [
+        BusyModule,
+        FormsModule,
+        ReactiveFormsModule,
+        RouterTestingModule.withRoutes([])
+      ],
+      providers: [
+        { provide: TasksService, useValue: tasksService },
+        { provide: ActivatedRoute, useValue: activeRoute },
+        { provide: ToastyService, useValue: toastyService }
+      ]
+    })
+      .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(TaskCreateComponent);
+    component = fixture.componentInstance;
```

Comment:
probably want to add `toastyService.clearAll();` to the beforeEach for future tests.